### ISSUE TITLE
Can I still use openSfM when images don't contain EXIF metadata such as focal length?

### ISSUE THREAD

**Issue by predictwise** [mapillary/OpenSfM#404]
I have downloaded the whole Berlin's images from Mapillary and want to reconstruct certain street by OpenSfM. 

I observe that OpenSfM needs to extract EXIF metadata from images, however, my image datasets don't contain any metadata expect for width and height. 

One of the most important metadata, focal length, is missing.

My question is: can I still use openSfM when images don't contain EXIF metadata such as focal length? If the metadata of Mapillary images is missing , how do you correct the misaligned photos to their original positions?

Looking forward to your replay!

**Comment by predictwise** [comment]
I have successfully corrected the misaligned photos to their original positions after some attempts. I just only used longitude and latitude.

Now, I want to measure the distance from the object on the photo to the camera. However, the focal length of the photo is unknown in that Mapillary doesn't offer the original photo for people to download.

My question is:  Where can I get the original photos, or how can I measure the distance under this circumstance?

**Comment by GITNE** [comment]
> My question is: can I still use openSfM when images don't contain EXIF metadata such as focal length?

Yes, you can. However, the output will be grossly inaccurate in the sense that proportions will match but distances will not. If no GPS data is available at all then OpenSfM will do reconstruct a point cloud but the resulting camera positions will have `0 N 0 E` (or close to) or no geographic coordinates. If any image lacks the `FocalLength` tag then OpenSfM assumes a default focal ratio (a quotient of focal plane width and focal length) of 0.85 which is roughly the mean focal ratio for most smartphones on the market.

> If the metadata of Mapillary images is missing , how do you correct the misaligned photos to their original positions?

Various tools exist to do this. One common tool combination is to use [JOSM](https://josm.openstreetmap.de) with the [Photo Adjust](https://wiki.openstreetmap.org/wiki/JOSM/Plugins/PhotoAdjust) plugin. You can query the [Mapillay API](https://a.mapillary.com) for the image's position and direction either in GeoJSON or GPX format. With that you can easily tag any images. Unfortunately, currently Mapillary's sequence GPX track export does not contain image directions but the GeoJSON response does.

> Where can I get the original photos

AFAIK, you have to acquire a commercial license. But, you should probably contact [sales](mailto:sales@mapillary.com) for any complete information. Or, you can contact the image author by leaving a comment on the image and ask them if they can provide you with the originals.

> how can I measure the distance under this circumstance?

This is not easy to answer. What kind of distance would you like to measure? If you know the size of the focal plane width and the focal length then using two overlapping photos you can measure basically every distance in the photos, including the distance between cameras. The Mapillary API reports the camera maker and model, hence you can look up the focal plane size (sensor width) and focal length (beware that the focal length may not be constant and thus vary with every image). Then you can add this metadata to your downloaded images.

**Comment by YanNoun** [comment]
No answer in a while, closing.
